cart/domain/cart/discount_test.go

```diff
@@ -423,6 +423,109 @@ func TestItem_HasDiscounts(t *testing.T) {
 	}
 }
 
```

Comment:
Please add a test with a complete cart including shipping and regular items to show that shipping items are now includes for the mergeDiscount on cart level